### PR TITLE
fix: sum udf change exception type & classify as USER error

### PR DESCRIPTION
**Pull request by bvarghese1**
### Description 
Throws KsqlFunctionException instead of ArithmeticException when there is an overflow for a
DECIMAL field using a SUM function. Additionally, it classifies KSqlFunctionException as a USER
error.

### Testing done 
Added unit and integration tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java [hunk 1]
```diff
@@ -258,7 +259,7 @@ private static void ensureMax(final BigDecimal value, final int precision, final
     final int digits = precision - scale;
     final BigDecimal maxValue = BigDecimal.valueOf(Math.pow(10, digits));
     if (maxValue.compareTo(value.abs()) < 1) {
-      throw new ArithmeticException(
```
Comment (spena) [inline ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java:261]: This ensureMax is an utility method that I think is called in different places other than functions, not just functions. In those cases, then KsqlFunctionException won't be a valid exception.

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  public KSqlFunctionClassifier(final String queryId) {
+    this.queryId = Objects.requireNonNull(queryId, "queryId");
+  }
+
+  @Override
+  public Type classify(final Throwable e) {
+    Type type = Type.UNKNOWN;
+    if (e instanceof KsqlFunctionException
+        || e instanceof StreamsException
+        && ExceptionUtils.getRootCause(e) instanceof KsqlFunctionException) {
```
Comment (jnh5y) [inline ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java:44]: I always get confused about the order of operations with `||` and `&&`.  I'd be tempted to add parens to make it completely clear.

Comment (bvarghese1) [reply ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java:44]: Sure, that makes sense. Will add parens to make it clear.

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  public KSqlFunctionClassifier(final String queryId) {
+    this.queryId = Objects.requireNonNull(queryId, "queryId");
+  }
+
+  @Override
+  public Type classify(final Throwable e) {
+    Type type = Type.UNKNOWN;
+    if (e instanceof KsqlFunctionException
+        || e instanceof StreamsException
```
[the remark this answers is not in the document]
Comment (bvarghese1) [reply ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java:43]: ```StreamsException``` is not being classified as a USER error. I will add parens to make that if condition easier to read. I have also added a test to validate the same.

File: ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java (hunk 2 above)
Comment (bvarghese1) [reply ksqldb-engine/src/main/java/io/confluent/ksql/query/KSqlFunctionClassifier.java:44]: Sure, that makes sense. Will add parens to make it clear.

File: ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java (hunk 1 above)
Comment (bvarghese1) [reply ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java:261]: @spena Thanks for pointing this out. After discussing with @jnh5y we have decided to revert this and make the change locally in the sum udaf.

**Pull request closed without merging**